Add date input `day`, `month`, `year` and `values` options (#6971)
alphagov/govuk-frontend: CHANGELOG.md

## Unreleased

### New features

> #### Add date input `day`, `month`, `year` and `values` options

Reviewer: I'd suggest waiting until @seaemsi has had a chance to review this too, but I think this should be user-focused rather than describe the change that we made.

So e.g. 'You can now pass `day`, `month`, `year`, `error` and `values` options to the date input component'

Author: Done 👍

> #### Add date input `day`, `month`, `year` and `values` options

Reviewer 2: ```suggestion
#### You can now pass `day`, `month`, `year`, `error` and `values` options to the Date input component
```

Author: Done 👍

> We've updated the date input component to add new Nunjucks options:

Reviewer 2: ```suggestion
We've updated the Date input component to add the following Nunjucks options:
```

Author: Done 👍

> - `day`, `month` and `year` to customise individual items

Reviewer 2: ```suggestion
- `day`, `month` and `year` options to customise individual items
```

Author: Done, thanks

- `error` boolean option to set the error state on individual items
- `values` option to set individual item values using a single object

> For consistency with other components, the date input error state is now set automatically when `errorMessage` is provided.

Reviewer 2: ```suggestion
For consistency with other components, the Date input component's error state is now set automatically when `errorMessage` is provided.
```

Author: Changed this, thanks
